editor/plugins/curve_editor_plugin.cpp

```diff
@@ -33,276 +33,356 @@
+			curve->set_meta(SNAME("_snap_enabled"), true);
+		} else {
+			curve->remove_meta(SNAME("_snap_enabled"));
+		}
+	}
+}
 
-	// Note: if you edit a curve, then set another, and try to undo,
-	// it will normally apply on the previous curve, but you won't see it
+void CurveEdit::set_snap_count(int p_snap_count) {
+	snap_count = p_snap_count;
+	queue_redraw();
+	if (curve.is_valid()) {
+		if (snap_count != CurveEditor::DEFAULT_SNAP) {
+			curve->set_meta(SNAME("_snap_count"), snap_count);
+		} else {
+			curve->remove_meta(SNAME("_snap_count"));
+		}
+	}
 }
 
-Size2 CurveEditor::get_minimum_size() const {
-	return Vector2(64, 150) * EDSCALE;
+Size2 CurveEdit::get_minimum_size() const {
+	return Vector2(64, 125) * EDSCALE;
 }
 
-void CurveEditor::_notification(int p_what) {
+void CurveEdit::_notification(int p_what) {
 	switch (p_what) {
+		case NOTIFICATION_ENTER_TREE:
+			connect("mouse_exited", callable_mp(this, &CurveEdit::_on_mouse_exited));
+			[[fallthrough]];
+		case NOTIFICATION_THEME_CHANGED:
+		case EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED: {
+			float gizmo_scale = EDITOR_GET("interface/touchscreen/scale_gizmo_handles");
+			point_radius = Math::round(BASE_POINT_RADIUS * get_theme_default_base_scale() * gizmo_scale);
+			hover_radius = Math::round(BASE_HOVER_RADIUS * get_theme_default_base_scale() * gizmo_scale);
+			tangent_radius = Math::round(BASE_TANGENT_RADIUS * get_theme_default_base_scale() * gizmo_scale);
+			tangent_hover_radius = Math::round(BASE_TANGENT_HOVER_RADIUS * get_theme_default_base_scale() * gizmo_scale);
+			tangent_length = Math::round(BASE_TANGENT_LENGTH * get_theme_default_base_scale());
+		} break;
 		case NOTIFICATION_DRAW: {
 			_draw();
 		} break;
-		case EditorSettings::NOTIFICATION_EDITOR_SETTINGS_CHANGED: {
-			if (EditorSettings::get_singleton()->check_changed_settings_in_group("interface/touchscreen/scale_gizmo_handles")) {
-				_gizmo_handle_scale = EDITOR_GET("interface/touchscreen/scale_gizmo_handles");
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (!is_visible()) {
+				grabbing = GRAB_NONE;
 			}
 		} break;
 	}
 }
 
-void CurveEditor::gui_input(const Ref<InputEvent> &p_event) {
-	Ref<InputEventMouseButton> mb_ref = p_event;
-	if (mb_ref.is_valid()) {
-		const InputEventMouseButton &mb = **mb_ref;
+void CurveEdit::gui_input(const Ref<InputEvent> &p_event) {
+	ERR_FAIL_COND(p_event.is_null());
+
+	Ref<InputEventKey> k = p_event;
+	if (k.is_valid()) {
+		// Deleting points or making tangents linear.
+		if (k->is_pressed() && k->get_keycode() == Key::KEY_DELETE) {
+			if (selected_tangent_index != TANGENT_NONE) {
+				toggle_linear(selected_index, selected_tangent_index);
+			} else if (selected_index != -1) {
+				if (grabbing == GRAB_ADD) {
+					curve->remove_point(selected_index); // Point is temporary, so remove directly from curve.
+					set_selected_index(-1);
+				} else {
+					remove_point(selected_index);
+				}
+				grabbing = GRAB_NONE;
+				hovered_index = -1;
```

Comment:
Yeah. Fixed.

Reply:
There's a couple more instances of this throughout the code, so worth checking for `hovered_index = -1` in general! :)

Reply:
I did- in this case here though the "fix" won't show because hovering is disabled anyway as the curve editor gets rebuild when stuff is removed. It's just visual though and also niche, so not that important. Maybe one day it'll get fixed on its own when that issue finally gets on the radar.